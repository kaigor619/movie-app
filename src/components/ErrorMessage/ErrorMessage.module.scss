.errorMessage {
  border: 0.15rem solid rgb(var(--coral-red-light-rgb));
  border-radius: 1rem;
  background-color: rgb(var(--background-error-rgb));
  text-align: center;
  padding: 1.5rem 0 2rem 0;
  max-width: 35rem;
  width: 100%;
  margin: auto;
}

.title {
  font-size: 1.7rem;
  margin: 0;
  color: var(--text-primary);
  font-weight: 600;
}

.image {
  max-width: 8rem;
  height: auto;
  width: 100%;
  align-self: center;
}

.description {
  color: rgba(var(--text-primary-rgb), 0.8);
  font-size: 0.9rem;
  padding: 0 1rem;
  margin: 1rem 0;
  margin: 0;
  margin-top: 1rem;
  font-weight: 300;
  text-align: left;
  display: inline-block;
}
