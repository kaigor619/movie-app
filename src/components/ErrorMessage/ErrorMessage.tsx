import styles from "./ErrorMessage.module.scss";
import brokenRobot from "./broken-robot.png";

interface Props {
  title?: string;
  description: string;
}

export const ErrorMessage: React.FC<Props> = ({
  title = "Error",
  description,
}) => {
  return (
    <div data-testid="error-message" className={styles.errorMessage}>
      <img className={styles.image} src={brokenRobot} alt="Broken robot" />
      <div className="">
        <h4 className={styles.title}>{title}</h4>
        <pre className={styles.description}>{description}</pre>
      </div>
    </div>
  );
};
