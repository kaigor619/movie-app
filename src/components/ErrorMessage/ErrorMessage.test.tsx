import { render, screen } from "@testing-library/react";
import { ErrorMessage } from "./ErrorMessage";

describe("ErrorMessage component", () => {
  it("applies default props correctly", () => {
    render(<ErrorMessage description="API doesn't work" />);
    expect(screen.getByText(/Error/i)).toBeInTheDocument();
  });
  it("shows props title and description", () => {
    const { asFragment } = render(
      <ErrorMessage title="My Error" description="API doesn't work" />
    );
    expect(screen.getByText(/Error/i)).toBeInTheDocument();
    expect(screen.getByText(/API doesn't work/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
  it("checks if the picture exists", () => {
    render(<ErrorMessage title="My Error" description="API doesn't work" />);
    expect(screen.getByRole("img")).toBeInTheDocument();
  });
});
