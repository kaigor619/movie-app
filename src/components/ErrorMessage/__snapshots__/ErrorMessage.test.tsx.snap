// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component shows props title and description 1`] = `
<DocumentFragment>
  <div
    class="errorMessage"
    data-testid="error-message"
  >
    <img
      alt="Broken robot"
      class="image"
      src="broken-robot.png"
    />
    <div
      class=""
    >
      <h4
        class="title"
      >
        My Error
      </h4>
      <pre
        class="description"
      >
        API doesn't work
      </pre>
    </div>
  </div>
</DocumentFragment>
`;
